### PR TITLE
Additional documentation & linting adjustments

### PR DESCRIPTION
**Pull request by stdoutput**
## Proposed Changes
- Adding additional documentation for exported client methods.
- Linting adjustments. These can be checked by using [golangci-lint](https://github.com/golangci/golangci-lint).

## Additional Information
I also tried to provide more meaningful names for some of the CFFI methods. I don't know how these may affect current CFFI users so feel free to remove them on review.

### CODE REVIEW

**Comment by bogdanfinn** [comment]
@stdoutput thank you very much for contributing. Really appreciate input here.
I will look at the changeset in detail later.

As far as i see you did not adjust the exported method names for the CFFI Methods? This should not affect current CFFI users when they just use the exported function names when working the the shared libraries.

File: client_options.go [hunk 1]
```diff
@@ -33,66 +33,83 @@ type httpClientConfig struct {
...
 }
 
+// WithNotFollowRedirects configures a HTTP client to not follow HTTP redirects.
 func WithNotFollowRedirects() HttpClientOption {
 	return func(config *httpClientConfig) {
 		config.followRedirects = false
 	}
 }
 
+// WithRandomTLSExtensionOrder configures a TLS client to randomize the order of TLS extensions being sent in the ClientHello.
```
Comment (bogdanfinn) [inline client_options.go:70]: GREASE and Padding Extension will stay in place. Only all other extensions will be randomized in order

File: cffi_src/types.go [hunk 2]
```diff
@@ -108,10 +110,11 @@ func (p *Timestamp) UnmarshalJSON(bytes []byte) error {
 		return fmt.Errorf("error decoding timestamp: %w", err)
 	}
 
-	*&p.Time = time.Unix(raw, 0)
+	p.Time = time.Unix(raw, 0)
 	return nil
 }
 
+// Response is the response that is sent back to the Python client.
```
Comment (bogdanfinn) [inline cffi_src/types.go:117]: Correct but i would like to see this comments more generalized and not focused on python itself as it is the same for node and every other language where users want to use the shared libraries

**Comment by stdoutput** [comment]
> @stdoutput thank you very much for contributing. Really appreciate input here. I will look at the changeset in detail later.
> 
> As far as i see you did not adjust the exported method names for the CFFI Methods? This should not affect current CFFI users when they just use the exported function names when working the the shared libraries.

Thanks for getting back so fast. I did not change the exported set, correct.

File: cffi_src/types.go (hunk 2 above)
Comment (stdoutput) [reply cffi_src/types.go:117]: Good point! I will adjust that.

**Pull request merged**